app/models/vm_or_template/operations/configuration.rb

```diff
@@ -74,7 +74,7 @@ def disconnect_floppies
   def raw_add_disk(disk_name, disk_size_mb, options = {})
     raise _("VM has no EMS, unable to add disk") unless ext_management_system
     if options[:datastore]
-      datastore = Storage.find_by(:name => options[:datastore])
+      datastore = ext_management_system.storages.find_by(:name => options[:datastore])
```

Comment:
@agrare hosts aren't available in this context. How would you suggest to do so ?

Also, the decision on which host to schedule the vm depends on auto-placement or not, in case of provision this is not in the context of this code.